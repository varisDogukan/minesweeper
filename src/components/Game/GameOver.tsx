import React, { FC } from "react";
import styled from "@emotion/styled";

export interface GameOverProps extends FrameProps {
  /**
   * Click handler
   */
  onClick: () => void;
}

export const GameOver: FC<GameOverProps> = ({ onClick, isWin }) => (
  <Frame onClick={onClick} isWin={isWin}>
    {isWin ? "😎" : "🙁"}
  </Frame>
);

interface FrameProps {
  /**
   * Is user win flag
   */
  isWin: boolean;
}

const Frame = styled.div<FrameProps>`
  top: 60%;
  left: 50%;
  z-index: 11;
  width: 8vw;
  height: 8vw;
  font-size: 4vw;
  text-align: center;
  user-select: none;
  cursor: pointer;
  line-height: 8vw;
  position: absolute;
  border-radius: 50%;
  pointer-events: auto;
  background-color: #d1d1d1;
  transform: translate(-50%, -50%);
  border: 1px solid rgba(51, 51, 51, 0.25);
`;
