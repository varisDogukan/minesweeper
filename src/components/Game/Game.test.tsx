import React from "react";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import { GameOver } from "./GameOver";

beforeEach(() => {
  jest.clearAllMocks();
});

describe("Game Over test cases", () => {
  it("GameOver render win correctly", async () => {
    const onReset = jest.fn();

    const { asFragment } = render(<GameOver onClick={onReset} isWin={true} />);

    const element = screen.getByText("ðŸ˜Ž");

    expect(element).toBeInTheDocument();

    await userEvent.click(element);

    expect(onReset).toHaveBeenCalled();

    expect(asFragment()).toMatchSnapshot();
  });

  it("GameOver render fail correctly", async () => {
    const onReset = jest.fn();

    const { asFragment } = render(<GameOver onClick={onReset} isWin={false} />);

    const element = screen.getByText("ðŸ˜“");

    expect(element).toBeInTheDocument();

    await userEvent.click(element);

    expect(onReset).toHaveBeenCalled();

    expect(asFragment()).toMatchSnapshot();
  });
});
