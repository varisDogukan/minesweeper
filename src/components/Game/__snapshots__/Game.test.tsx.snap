// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game Over test cases GameOver render fail correctly 1`] = `
<DocumentFragment>
  <div
    class="css-1uy3707"
  >
    ðŸ˜“
  </div>
</DocumentFragment>
`;

exports[`Game Over test cases GameOver render win correctly 1`] = `
<DocumentFragment>
  <div
    class="css-1uy3707"
  >
    ðŸ˜Ž
  </div>
</DocumentFragment>
`;
